python_modules/dagster-ext/dagster_ext/__init__.py

```diff
@@ -543,6 +543,80 @@ def upload_messages_chunk(self, payload: IO, index: int) -> None:
 # ########################
 
 
+class IContext(ABC):
+    """Base class for asset context implemented by AssetExecutionContext and ExtContext."""
+
+    @property
+    @abstractmethod
+    def is_asset_step(self) -> bool:
+        """TODO."""
+
+    @property
+    @abstractmethod
+    def asset_key(self) -> str:
+        """TODO."""
+
+    @property
+    @abstractmethod
+    def asset_keys(self) -> Sequence[str]:
+        """TODO."""
+
+    @property
+    @abstractmethod
+    def provenance(self) -> Optional[ExtDataProvenance]:
+        """TODO."""
+
+    @property
+    @abstractmethod
+    def provenance_by_asset_key(self) -> Mapping[str, Optional[ExtDataProvenance]]:
+        """TODO."""
+
+    @property
+    @abstractmethod
+    def code_version(self) -> Optional[str]:
+        """TODO."""
+
+    @property
+    @abstractmethod
+    def code_version_by_asset_key(self) -> Mapping[str, Optional[str]]:
+        """TODO."""
+
+    @property
+    @abstractmethod
+    def is_partition_step(self) -> bool:
+        """TODO."""
+
+    @property
+    @abstractmethod
+    def partition_key(self) -> str:
+        """TODO."""
+
+    @property
+    @abstractmethod
+    def partition_key_range(self) -> Optional["ExtPartitionKeyRange"]:
+        """TODO."""
+
+    @property
+    @abstractmethod
+    def partition_time_window(self) -> Optional["ExtTimeWindow"]:
+        """TODO."""
+
```

Comment:
TODO - investigate if the above methods make working with StaticPartitions difficult/impossible since there is no "range" of keys. Perhaps just `partition_key` is enough, but need to verify

Reply:
might just need to add the `partition_keys` method to the context

Reply:
partition methods punted until https://github.com/dagster-io/internal/discussions/6704 is resolved. I'll add the partition methods to the interface once we decide what they should be